File: 2570-merge-two-2d-arrays-by-summing-values/2570-merge-two-2d-arrays-by-summing-values.cpp
class Solution {
public:
    vector<vector<int>> mergeArrays(vector<vector<int>>& nums1, vector<vector<int>>& nums2) {
        
        vector<vector<int>>ans;
        map<int,int>m;
        
        for(auto arr : nums1){
            if(m[arr[0]])
                m[arr[0]] += arr[1];
            else
                m[arr[0]] = arr[1];
        }
        
        for(auto arr : nums2){
            if(m[arr[0]])
                m[arr[0]] += arr[1];
            else
                m[arr[0]] = arr[1];
        }
        
        for(auto x : m){
            ans.push_back({x.first,x.second});
        }
        
        return ans;
    }
};

File: 2570-merge-two-2d-arrays-by-summing-values/2570-merge-two-2d-arrays-by-summing-values.ts
function mergeArrays(nums1: number[][], nums2: number[][]): number[][] {
    let map = {};
    let res = [];
    
    nums1.forEach((arr)=>{
        map[arr[0]] = map[arr[0]] ? map[arr[0]] + arr[1] : arr[1];
    })
    nums2.forEach((arr)=>{
        map[arr[0]] = map[arr[0]] ? map[arr[0]] + arr[1] : arr[1];
    })
    
    Object.keys(map).map((key)=>{
        res.push([key,map[key]])
    })
    return res;
};